docs/proposals/rescheduling/README.md

```diff
@@ -0,0 +1,273 @@
+---
+title: Rescheduling mechanism based on application health status
+authors:
+  - "@Poor12"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+
+create-date: 2023-02-10
+
+---
+# Rescheduling mechanism for unhealthy application state
+
+## Summary
+
+In the multi-cluster scenario, user workloads may be deployed in multiple clusters to improve service high availability. Now Karmada already supports multi-cluster failover when detecting a cluster fault.
+However, the application may still be unavailable when the cluster is in a healthy state. For example, the resources of the cluster are reclaimed so that the application has no resources to run.
+Therefore, Karmada needs to provide a means of fault migration from an application perspective.
+
+## Motivation
+
+The cluster administrators want to ensure that the application can run normally, regardless of which cluster the application is running. When the application is unavailable,
+they want Karmada to reschedule the application and avoid clusters that failed to run before.
+
+## Goals
+
+- Extend the API of PropagationPolicy to provide users with the rescheduled configuration.
+- Extend the API of ResourceBinding to record previously evicted clusters.
+- Propose the implementation ideas for involved components, including `karmada-controller-manager`, `karmada-webhook` and `karmada-scheduler`.
+For example, `karmada-controller-manager` need to add a component to observe the health status of the application. `Karmada-scheduler` need to
+add a plugin to filter the evicted cluster.
+
+## Non-Goals
+
+## Proposal
+
+### User Stories (Optional)
+
+#### As a user, I use the cloud vendor's spot instance to deploy my application. As a result, the application may fail due to insufficient resources.
+
+Spot instances are a mode of instances that feature a discounted price and a system interruption mechanism, which means that you can purchase a spot instance ar a discounted price, but the system may automatically repossess it.
+When users use spot instances to deploy applictions, the probability of an application failing in a vendor's cluster increases significantly. This can also occur in clusters with resource preemption and reclamation.
```

Comment:
> When users use spot instances to deploy applictions, the probability of an application failing in a vendor's cluster increases significantly.

That doesn't make sense, why use a spot instance would `increase` failing probability?

Reply:
> This can also occur in clusters with resource preemption and reclamation.

This is another use case, not `spot instance`, right? Then tell the story in another story.

Reply:
> > This can also occur in clusters with resource preemption and reclamation.
> 
> This is another use case, not `spot instance`, right? Then tell the story in another story.

This is a mistake. The key is clusters with a dynamically resource pool. If the resource pool is stable, estimator will get the accurate resource margin.